lib/util/http/gerrit.ts

```diff
@@ -0,0 +1,100 @@
+import JSON5 from 'json5';
+import { ZodSchema } from 'zod';
+import type { HttpOptions, HttpResponse, InternalHttpOptions } from './types';
+import { Http } from './index';
+
+let baseUrl: string;
+export const setBaseUrl = (url: string): void => {
+  baseUrl = url;
+};
+
+/**
+ * Access Gerrit REST-API and strip-of the "magic prefix" from responses.
+ * @see https://gerrit-review.googlesource.com/Documentation/rest-api.html
+ */
+export class GerritHttp extends Http {
+  magicPrefix = /^\)]}'\n/g;
+
+  constructor(options?: HttpOptions) {
+    super('gerrit', options);
+  }
+
+  protected override request<T>(
+    path: string,
+    options?: InternalHttpOptions
+  ): Promise<HttpResponse<T>> {
+    const url = baseUrl + path;
+    const opts = {
+      baseUrl,
+      ...options,
+    };
+    opts.headers = {
+      ...opts.headers,
+    };
+    return super.request<T>(url, opts);
+  }
+
+  override get(url: string, options: HttpOptions = {}): Promise<HttpResponse> {
+    return super.get(url, options).then((res) => ({
+      ...res,
+      body: res.body.replaceAll(this.magicPrefix, ''),
+    }));
+  }
+
+  override getJson<T = any>(
+    url: string,
+    options?: HttpOptions | ZodSchema,
+    arg3?: ZodSchema
+  ): Promise<HttpResponse<T>> {
+    // istanbul ignore next
+    const httpOptions = options instanceof ZodSchema ? undefined : options;
+    return super.get(url, httpOptions).then((res) => ({
+      ...res,
+      body: JSON5.parse(res.body.replaceAll(this.magicPrefix, '')),
+    }));
+  }
```

Comment:
Do this inside request, so it works for all responses, use the response content-type header to limit to json responses. That way you shouldn't require to override all those functions

Reply:
done